python/cudf/cudf/utils/dateutils.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+
+import calendar
+
+from cudf.utils.dtypes import is_scalar
+
+day_names = list(calendar.day_name)
+month_names = list(calendar.month_name)
+
+
+def wkday_idx_to_name(inds_in):
+    indices = range(len(day_names) + 1)
+    ind_to_day = dict(zip(indices, day_names))
+    if is_scalar(inds_in):
+        return ind_to_day[inds_in]
+    else:
+        return [ind_to_day[i] for i in inds_in]
```

Comment:
This is problematic if `inds_in` is a column-like object. Instead we likely need a join but need to maintain the original order. This sounds similar to what we had to do for categoricals.